demo_nodes_cpp/include/demo_nodes_cpp/visibility_control.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DEMO_NODES_CPP__VISIBILITY_CONTROL_H_
```

Comment:
C++ headers should have the extension `.hpp`.

Rename this file and the other visibility control files

Reply:
The visibility control header is C code - not C++.

Reply:
True. I was just looking at the [rclcpp visibility header](https://github.com/ros2/rclcpp/blob/9be3e08cd4d04193fb76cb6dd8f5db11d7199e2f/rclcpp/include/rclcpp/visibility_control.hpp), which has the `.hpp` suffix. Since we're not writing C code in the demo packages and, AFAIU, the visibility macros are for C++ code, would it be wrong to drop the `#ifdef __cplusplus` stuff and rename the file to `visibility_control.hpp`?